src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -378,3 +385,40 @@ int enroll_tpm2(struct crypt_device *cd,
         log_info("New TPM2 token enrolled as key slot %i.", keyslot);
         return keyslot;
 }
+
+int vl_method_enroll_tpm2(Varlink *link, JsonVariant *params, VarlinkMethodFlags flags, void *userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                VARLINK_DISPATCH_UNLOCK_FIELDS,
+                VARLINK_DISPATCH_WIPE_FIELDS,
+                VARLINK_DISPATCH_POLKIT_FIELD,
+                {}
+        };
+        Hashmap **polkit_registry = ASSERT_PTR(userdata);
+        _cleanup_(crypt_freep) struct crypt_device *cd = NULL;
+        _cleanup_(erase_and_freep) void *vk = NULL;
+        size_t vks;
+        int r;
+
+        r = varlink_dispatch(link, params, dispatch_table, NULL);
+        if (r != 0)
+                return r;
+
+        r = varlink_verify_polkit_async(
+                        link,
+                        /* bus= */ NULL,
+                        "io.systemd.cryptenroll.enroll",
+                        /* details= */ NULL,
+                        /* good_user= */ UID_INVALID,
+                        polkit_registry);
+        if (r <= 0)
+                return r;
+
+        r = vl_luks_setup(link, params, &cd, &vk, &vks);
+        if (r != 0)
+                return r;
+
+        return varlink_errorb(
+                        link,
+                        VARLINK_ERROR_METHOD_NOT_IMPLEMENTED,
+                        JSON_BUILD_OBJECT(JSON_BUILD_PAIR("method", "io.systemd.CryptEnroll.EnrollTPM2")));
```

Comment:
might be a candidate for a high-level helper:

varlink_error_method_not_implemented() 


(similar to how we have varlink_error_invalid_parameter() and varlink_error_invalid_parameter_name())